pkg/api/v1alpha1/machine_types.go

```diff
@@ -13,3 +15,30 @@ type UserConfiguration struct {
 	Name              string   `json:"name"`
 	SshAuthorizedKeys []string `json:"sshAuthorizedKeys"`
 }
+
+func defaultMachineConfigUsers(users []UserConfiguration) []UserConfiguration {
+	if len(users) <= 0 {
+		users = []UserConfiguration{{}}
+	}
+	if len(users[0].SshAuthorizedKeys) <= 0 {
+		users[0].SshAuthorizedKeys = []string{""}
+	}
+	if users[0].Name == "" {
+		users[0].Name = DefaultCloudStackUser
```

Comment:
Ah, you're right. This was a mistake. Here's I'm trying to dry out this logic because it will be similar in other providers. I'll make the necessary adjustment